subsys/random/rand32_timer.c

```diff
@@ -41,7 +58,7 @@ static atomic_val_t _rand32_counter;
  */
 uint32_t z_impl_sys_rand32_get(void)
 {
-	return k_cycle_get_32() + atomic_add(&_rand32_counter, _RAND32_INC);
+	return k_cycle_get_32() + xoshiro128_next();
```

Comment:
As we are now using the quality pseudo-random algorithm - let's drop this `k_cycle_get_32` which was used as an attempt of workaround the previous badly designed implementation.

Reply:
Agree with @dleach02 here. It is import that we distinguish between "quality" of a pseudo-random algorithm for things like network timing backoffs and similar, and a cryptographic quality PRNG, which xoshiro128 is not.

Reply:
> Agree with @dleach02 here. It is import that we distinguish between "quality" of a pseudo-random algorithm for things like network timing backoffs and similar

And I'm indeed talking about that :)

We already have multiple re-spins of rand32_timer which were trying to fix some situation when rand32_timer produce non random output (e.g see #37499) but we still have issues with it because of this k_cycle_get_32() addition - like in the current #47908 issue. It's not about cryptographic quality PRNG - it's about PRNG producing same value every call :)
Probably it' time to drop this k_cycle_get_32 call.

And as for naming - I guess it's fine to rename it to `rand32_test.c` as it anyway depends on `TEST_RANDOM_GENERATOR` option.